### PR TITLE
Adds release notes for 2.1.0 and revert #1890

### PR DESCRIPTION
**Pull request by DarshitChanpura**
### Description

* Category - Maintenance

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-security.release-notes-2.1.0.0.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+### Maintenance
+* Revert "Bump version to 2.1.0.0 (#1865)" ([#1882](https://github.com/opensearch-project/security/pull/1882))
+* Bump version to 2.1.0.0 ([#1865](https://github.com/opensearch-project/security/pull/1865))
+* Revert "Bump version to 2.1.0.0 (#1855)" ([#1864](https://github.com/opensearch-project/security/pull/1864))
+* Bump version to 2.1.0.0 ([#1855](https://github.com/opensearch-project/security/pull/1855))
+* Add suppression for all removal warnings ([#1828](https://github.com/opensearch-project/security/pull/1828))
+* Update support link ([#1851](https://github.com/opensearch-project/security/pull/1851))
+* Create 2.0.0 release notes ([#1854](https://github.com/opensearch-project/security/pull/1854))
+* Switch to standard OpenSearch gradle build ([#1888](https://github.com/opensearch-project/security/pull/1888))
+* Adds default roles for Snapshot Management plugin ([#1897](https://github.com/opensearch-project/security/pull/1897))
```
Comment (peternied) [inline release-notes/opensearch-security.release-notes-2.1.0.0.md:22]: This seems like an enhancement

File: release-notes/opensearch-security.release-notes-2.1.0.0.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+### Enhancements
+* Delegate to NettyAllocator.getAllocator() for ByteBufAllocator instead of hard-coding PooledByteBufAllocator. ([#1396](https://github.com/opensearch-project/security/pull/1396))
+* Tenant Permissions : added the possibility to specify tenants via parameter ([#1813](https://github.com/opensearch-project/security/pull/1813))
+* JWT: validate issuer and audience ([#1780](https://github.com/opensearch-project/security/pull/1780), [#1781](https://github.com/opensearch-project/security/pull/1781)) ([#1785](https://github.com/opensearch-project/security/pull/1785))
+
+### Refactoring
+* Remove master keywords ([#1886](https://github.com/opensearch-project/security/pull/1886))
+
+### Maintenance
```
Comment (peternied) [inline release-notes/opensearch-security.release-notes-2.1.0.0.md:13]: IMO we should delete this section, our customers that read release notes aren't interested in version bumping or release note related pull requests - if they were they should check our our git log instead.

If we do want to keep this section, please move it to the bottom of the document so we can highlight the bug fixes more prominently

File: release-notes/opensearch-security.release-notes-2.1.0.0.md (hunk 1 above)
Comment (DarshitChanpura) [reply release-notes/opensearch-security.release-notes-2.1.0.0.md:32]: done

File: release-notes/opensearch-security.release-notes-2.1.0.0.md (hunk 2 above)
Comment (DarshitChanpura) [reply release-notes/opensearch-security.release-notes-2.1.0.0.md:23]: makes sense... I have moved it to the bottom for now

**Review by peternied — APPROVED** [review]

**Comment by cliu123** [comment]
Converting the draft to an open PR for review.

**Review by cliu123 — APPROVED** [review]

**Review by peternied — DISMISSED** [review]

**Comment by cliu123** [comment]
@DarshitChanpura Why this release notes PR includes other changes? If those changes have been merged, please rebase this branch.

**Review by peternied — APPROVED** [review]

**Review by cliu123 — APPROVED** [review]

**Pull request merged**